### PR TITLE
Feature/60 fluent bit

### PR DESCRIPTION
**Pull request by kcoleman731**
*Issue #, if available:*

*Description of changes:*

Adds `aws-for-fluent-bit` as an add-on. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: lib/addons/aws-for-fluent-bit/index.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        super({ ...defaultProps as any, ...props })
+        this.options = this.props
+    }
+
+    deploy(clusterInfo: ClusterInfo): Promise<Construct> {
+        const cluster = clusterInfo.cluster;
+
+        // Create the FluentBit namespace.
+        const namespace = this.options.namespace
+        createNamespace(this.options.namespace!, cluster)
```
Comment (shapirov103) [inline lib/addons/aws-for-fluent-bit/index.ts:48]: The CI run caught a failure here: if namespace is left as kube-system this call fails. One way is to check for namespace name, but we could also add a flag, like in VeleroAddOn `createNamespace: boolean` to make it explicit. And that one could false by default.

File: docs/addons/aws-for-fluent-bit.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
+# AWS for Fluent Bit
+
+Fluent Bit is an open source Log Processor and Forwarder which allows you to collect any data like metrics and logs from different sources, enrich them with filters and send them to multiple destinations.
```
Comment (askulkarni2) [inline docs/addons/aws-for-fluent-bit.md:3]: Suggest rewording as "Fluent Bit is an open source <strike>L</strike>log processor and <strike>F</strike>forwarder which allows you to collect <strike>any</strike> data  such as metrics and logs from different sources... "

File: lib/addons/aws-for-fluent-bit/index.ts (hunk 1 above)
Comment (askulkarni2) [reply lib/addons/aws-for-fluent-bit/index.ts:48]: Shouldn't the helper function be updated to do a no op if namespace exists?

**Review by askulkarni2 — COMMENTED** [review]
waiting for CI tests to pass, looks good otherwise.

File: lib/addons/aws-for-fluent-bit/index.ts (hunk 1 above)
Comment (shapirov103) [reply lib/addons/aws-for-fluent-bit/index.ts:48]: Good point! The createnamespace function should be leveraging kubetl apply. In CDK it is overwrite flag:

```
ew KubernetesManifest(cluster.stack, name, {
            cluster,
            manifest,
            overwrite: true
        })
```

File: lib/addons/aws-for-fluent-bit/index.ts [hunk 3]
```diff
@@ -0,0 +1,73 @@
+import { PolicyStatement } from '@aws-cdk/aws-iam';
+import { Construct } from "@aws-cdk/core";
+
+import { HelmAddOn, HelmAddOnUserProps } from "../helm-addon";
+import { ClusterInfo } from "../../spi/types";
+import { createNamespace } from "../../utils/namespace-utils";
+
+/**
+ * Configuration options for the FluentBit add-on.
```
Comment (shapirov103) [inline lib/addons/aws-for-fluent-bit/index.ts:9]: There is a ton of options that this addon takes, e.g. cloudwatch forwarding, kinesis forwarding, es, etc. 
It would help to add a link to the values supported by the add-on. It can be done here, or by adding values field that matches the definition of the super interface and adding docs section to it.

**Review by shapirov103 — COMMENTED** [review]
Looks good, a couple of comments.

File: lib/addons/aws-for-fluent-bit/index.ts (hunk 3 above)
Comment (kcoleman731) [reply lib/addons/aws-for-fluent-bit/index.ts:9]: Done. Added to doc block for the add-on

File: lib/addons/aws-for-fluent-bit/index.ts [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+/**
+ * AwsForFluentBitAddOn deploys FluentBit into an EKS cluster using the `aws-for-fluent-bit` Helm chart.
+ * https://github.com/aws/eks-charts/tree/master/stable/aws-for-fluent-bit
+ * 
+ * For information on how to configure the `aws-for-fluent-bit` Helm chart to forward logs and metrics to AWS services like CloudWatch or Kinesis, please view the values.yaml spec provided by the chart.
+ * https://github.com/aws/eks-charts/blob/master/stable/aws-for-fluent-bit/values.yaml#L54
+ */
+export class AwsForFluentBitAddOn extends HelmAddOn {
+
+    readonly options: AwsForFluentBitAddOnProps
```
Comment (shapirov103) [inline lib/addons/aws-for-fluent-bit/index.ts:39]: minor: let's address the missing semi-colon (`make lint` should catch it if merged with main)

File: lib/addons/aws-for-fluent-bit/index.ts [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+import { createNamespace } from "../../utils/namespace-utils";
+
+/**
+ * Configuration options for the FluentBit add-on.
+ */
+export interface AwsForFluentBitAddOnProps extends HelmAddOnUserProps {
+    /**
+     * Iam policies for the add-on.
+     */
+    iamPolicies?: PolicyStatement[]
```
Comment (shapirov103) [inline lib/addons/aws-for-fluent-bit/index.ts:15]: We need an issue to create a pattern for at least one case when this field is populated with policy per resource (e.g. access to CW or ES).

File: docs/addons/aws-for-fluent-bit.md [hunk 6]
```diff
@@ -0,0 +1,22 @@
+# AWS for Fluent Bit
+
+Fluent Bit is an open source log processor and forwarder which allows you to collect data like metrics and logs from different sources, enrich them with filters and send them to multiple destinations.
+
+AWS provides a Fluent Bit image with plugins for both CloudWatch Logs and Kinesis Data Firehose. The [AWS for Fluent Bit image](https://gallery.ecr.aws/aws-observability/aws-for-fluent-bit) is available on the Amazon ECR Public Gallery. For more details, see [AWS for Fluent Bit GitHub repository](https://github.com/aws/aws-for-fluent-bit).
+
+## Usage
+
```
Comment (shapirov103) [inline docs/addons/aws-for-fluent-bit.md:8]: Can we add a section on Configuration and reference the values from the charts that you referenced in the docs (https://github.com/aws/eks-charts/tree/master/stable/aws-for-fluent-bit) in the doc as well, similar to what we did for e.g. Nginx[here](https://github.com/aws-quickstart/ssp-amazon-eks/blob/main/docs/addons/nginx.md#configuration)?

**Review by shapirov103 — COMMENTED** [review]
A couple of comments, mostly docs related. Looks good!

**Comment by shapirov103** [comment]
@kcoleman731 `make lint` gives 20 warnings (semicolon). I will autofix, but this setting should be active in your workspace.

File: lib/addons/index.ts [hunk 7]
```diff
@@ -1,6 +1,27 @@
...
+export * from './metrics-server'
+export * from './nginx'
+export * from './opa-gatekeeper'
+export * from './xray'
+export * from './secrets-store'
+export * from './secrets-store/secret-provider'
+export * from './ssm-agent'
+export * from './nested-stack'
+export * from './velero'
+export * from './vpc-cni'
```
Comment (shapirov103) [inline lib/addons/index.ts:19]: multiple duplicates

**Review by shapirov103 — APPROVED** [review]
lgtm
